### PR TITLE
Refactor the eventhubs binder health indicator

### PR DESCRIPTION
**Pull request by moarychan**
Fixes Azure/azure-sdk-for-java#25226

This implementation refers to the Service Bus Binder, which is more suitable for developers.

TODO:
Refactor the unittests.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/main/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinder.java [hunk 1]
```diff
@@ -122,7 +137,9 @@ protected MessageProducer createConsumerEndpoint(ConsumerDestination destination
                 destination.getName(), group, properties.getExtension().getCheckpoint());
         }
         inboundAdapter.setBeanFactory(getBeanFactory());
-
+        String instrumentationId = Instrumentation.buildId(CONSUMER, destination.getName() + "/" + (!INVALID_SUBSCRIPTION.equals(group) ? group : ""));
```
Comment (yiliuTo) [inline sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/main/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinder.java:140]: for eventhubs, the consumer group is a required option, so we can ignore the case of a invalid group.

File: sdk/spring/spring-integration-azure-eventhubs/src/main/java/com/azure/spring/integration/eventhubs/inbound/health/EventHusProcessorInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import com.azure.messaging.eventhubs.models.ErrorContext;
+import com.azure.spring.integration.instrumentation.Instrumentation;
+
+import java.time.Duration;
+
+/**
+ * EventHus health details entity class.
+ */
+public class EventHusProcessorInstrumentation implements Instrumentation {
```
Comment (yiliuTo) [inline sdk/spring/spring-integration-azure-eventhubs/src/main/java/com/azure/spring/integration/eventhubs/inbound/health/EventHusProcessorInstrumentation.java:14]: Can we combine the EventHusProcessorInstrumentation and ServiceBusProcessorInstrumentation as a ProcessorInstrumentation?

Comment (moarychan) [reply sdk/spring/spring-integration-azure-eventhubs/src/main/java/com/azure/spring/integration/eventhubs/inbound/health/EventHusProcessorInstrumentation.java:14]: Most can be merged, but the difference is in `Throwable getException();` method.

File: sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/main/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinder.java [hunk 3]
```diff
@@ -199,16 +192,31 @@ public String getConsumerGroup() {
         };
     }
 
-    private EventHubsTemplate getEventHubTemplate() {
+    public EventHubsTemplate getEventHubTemplate() {
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/main/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinder.java:195]: why change this to public

File: sdk/spring/spring-integration-azure-eventhubs/src/main/java/com/azure/spring/integration/eventhubs/inbound/EventHubsInboundChannelAdapter.java [hunk 4]
```diff
@@ -74,10 +78,10 @@ public EventHubsInboundChannelAdapter(EventHubsProcessorContainer eventProcessor
 
     @Override
     protected void onInit() {
-        if (ListenerMode.RECORD.equals(this.listenerMode)) {
-            this.listener = recordEventProcessor;
-        } else {
+        if (ListenerMode.BATCH.equals(this.listenerMode)) {
             this.listener = batchEventProcessor;
+        } else {
+            this.listener = recordEventProcessor;
```
Comment (saragluna) [inline sdk/spring/spring-integration-azure-eventhubs/src/main/java/com/azure/spring/integration/eventhubs/inbound/EventHubsInboundChannelAdapter.java:84]: why change this order

**Review by saragluna — APPROVED** [review]
LGTM

**Pull request merged**